### DISCUSSION TITLE
Need some help with adding a zigbee-herdsman converter for my custom device

### DISCUSSION THREAD

**Discussion by Tokn59** [Koenkk/zigbee2mqtt#20302] [Devices]
Hi, I successfully created a zigbee device that is recognised in zigbee2mqtt. It contains a DHT22 sensor for temperature and humidity readings and also a button connected to a GPIO pin (10). The temp & hum readings are working fine but I can't get the button to zigbee2mqtt. It seems I need to define a converter for a binary input of some sorts.

Reading the cluster works : Read result of 'genBinaryInput': {"presentValue":1}

This is my converter : 

<img width="847" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/9215231/55b9aff5-9a53-4fc0-b64a-5cca28433e63">

"switch" does not represent the status of the binary input on GPIO10. Could someone please help me "reading" the presentvaluestatus ? Big thanks for your help !!!!!

**Comment by bandit7311** [comment]
@Tokn59  On which platform did you program , maybe a ESP32-C6?

Thanks B
